### ISSUE TITLE
methods do not work on the returned objects from dataset.getTables

### ISSUE THREAD

**Issue by nikhilbobb** [GoogleCloudPlatform/gcloud-node#1114]
On calling dataset.getTables from the bigquery API, and then calling any method on the returned tables I get the following error:

Invalid table ID

Am I doing something wrong?

**Comment by stephenplusplus** [comment]
I was able to reproduce as well. Looks like table IDs are returned from the upstream APIs like this:

```
nth-circlet-705:gcloud_test_dataset_tempc24392d0_cbac_11e5_9063_0923b7e0f6f8.myKittens
```

I'm not sure if it's always been this way, or if we just missed it. But what we want instead is:

```js
apiResponse.tableReference.tableId;
```

In the meantime, you can use something like this code to get around the bug:

```js
dataset.getTables(function(err, tables) {
  tables = tables.map(function(table) {
    var fixedTable = dataset.table(table.metadata.tableReference.tableId);
    fixedTable.metadata = table.metadata;
    return fixedTable;
  });
});
```

Thanks for catching this and reporting it!

**Comment by nikhilbobb** [comment]
Thanks Stephen!

**Comment by stephenplusplus** [comment]
Thank you! @callmehiphop gave us the fix in #1116. We'll hopefully be releasing on Monday, keep an eye out!
